model/src/main/kotlin/config/AnalyzerConfiguration.kt

```diff
@@ -36,4 +36,9 @@ data class AnalyzerConfiguration(
      * false.
      */
     val allowDynamicVersions: Boolean = false,
+
+    /**
+     * Configuration of the SW360 package curation provider.
+     */
+    val sw360Configuration: Sw360StorageConfiguration? = null
```

Comment:
Due to this, this commit also needs to add `sw360_configuration: null` to the `config` section of expcted ORT result files to make tests pass. This should be the list of files:

```
$ git grep -F "    ignore_tool_versions:" -- "*.yml"
analyzer/src/funTest/assets/projects/external/directories-expected-output.yml:    ignore_tool_versions: false
analyzer/src/funTest/assets/projects/external/git-repo-expected-output.yml:    ignore_tool_versions: false
analyzer/src/funTest/assets/projects/external/sbt-multi-project-example-expected-output.yml:    ignore_tool_versions: false
analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml:    ignore_tool_versions: false
analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result.yml:    ignore_tool_versions: false
cli/src/funTest/assets/semver4j-analyzer-result.yml:    ignore_tool_versions: false
reporter/src/funTest/assets/gradle-all-dependencies-result.yml:    ignore_tool_versions: false
reporter/src/funTest/assets/static-html-reporter-test-input.yml:    ignore_tool_versions: false
scanner/src/funTest/assets/analyzer-result.yml:    ignore_tool_versions: false
scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml:    ignore_tool_versions: false
```

Reply:
I think it should rather not serialize this property if it is null instead.

Reply:
Agreed, that's probably better. Sorry @bs-ondem for me not coming up with that idea in the first place.